services/QuillLMS/app/queries/quill_big_query/concerns/generated_view_query_fallback.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module QuillBigQuery
+  module Concerns
+    module GeneratedViewQueryFallback
+      extend ActiveSupport::Concern
+
+      class BrokenViewError < StandardError; end
+
+      BROKEN_VIEW_ERRORS = [
+        ::Google::Cloud::NotFoundError,
+        ::Google::Cloud::InvalidArgumentError
+      ]
+
+      def run_query = run_query_with_fallback
+
+      def run_query_with_fallback
+        query_runner(query)
+      rescue *BROKEN_VIEW_ERRORS => e
```

Comment:
Are these the only probable emitted errors at this path in the execution? If not, consider a more liberal rescue matcher.

Reply:
When we did our original testing for fallback logic, these were the only two errors that were emitted in cases where the referenced Materialized Views were unavailable.